cpp/tests/reductions/scan_tests.cpp

```diff
@@ -0,0 +1,476 @@
+
+    std::partial_sum(
+        b.cbegin(),
+        b.cend(),
+        out_b.begin(),
+        std::logical_and<bool>{});
+
+    // string column with nulls
+    cudf::test::strings_column_wrapper col_nulls(v.begin(), v.end(), b.begin());
+    cudf::test::strings_column_wrapper expected2(exact.begin(), exact.end(), out_b.begin());
+    std::unique_ptr<cudf::column> col_out;
+    //skipna=false
+    CUDF_EXPECT_NO_THROW(col_out = cudf::experimental::scan(col_nulls,
+        cudf::experimental::make_max_aggregation(), scan_type::INCLUSIVE, include_nulls::YES));
+    if(do_print) {
+        print_view(expected2, "expect = ");
+        print_view(col_out->view(),   "result = ");
+    }
+    cudf::test::expect_column_properties_equal(expected2, col_out->view());
+    cudf::test::expect_columns_equal(expected2, col_out->view());
+
+    //Exclusive scan string not supported.
+    CUDF_EXPECT_THROW_MESSAGE((cudf::experimental::scan(col_nulls,
+    cudf::experimental::make_min_aggregation(), scan_type::EXCLUSIVE, include_nulls::NO)),
+    "String types supports only inclusive min/max for `cudf::scan`");
+
+    CUDF_EXPECT_THROW_MESSAGE((cudf::experimental::scan(col_nulls,
+    cudf::experimental::make_min_aggregation(), scan_type::EXCLUSIVE, include_nulls::YES)),
+    "String types supports only inclusive min/max for `cudf::scan`");
+    }
+
+TYPED_TEST(ScanTest, EmptyColumnskip_nulls)
+{
+  bool do_print=false;
+  std::vector<TypeParam> v{};
+  std::vector<bool>      b{};
+  cudf::test::fixed_width_column_wrapper<TypeParam> const col_in{v.begin(), v.end(),
+                                                            b.begin()};
+  std::unique_ptr<cudf::column> col_out;
+  
+  //test output calculation
+  std::vector<TypeParam> out_v(v.size());
+  std::vector<bool>      out_b(v.size());
+  
+  //skipna=true (default)
+  CUDF_EXPECT_NO_THROW(col_out = cudf::experimental::scan(col_in, 
+    cudf::experimental::make_sum_aggregation(), scan_type::INCLUSIVE, include_nulls::NO));
+  cudf::test::fixed_width_column_wrapper<TypeParam> expected_col_out1{
+      out_v.begin(), out_v.end(), b.cbegin()};
+  cudf::test::expect_column_properties_equal(expected_col_out1, col_out->view());
+  cudf::test::expect_columns_equal(expected_col_out1, col_out->view());
+  if(do_print) {
+    print_view(expected_col_out1, "expect = ");
+    print_view(col_out->view(),   "result = ");
+  }
+
+  //skipna=false
+  CUDF_EXPECT_NO_THROW(col_out = cudf::experimental::scan(col_in, 
+  cudf::experimental::make_sum_aggregation(), scan_type::INCLUSIVE, include_nulls::YES));
+  cudf::test::fixed_width_column_wrapper<TypeParam> expected_col_out2{
+      out_v.begin(), out_v.end(), out_b.begin()};
+  if(do_print) {
+    print_view(expected_col_out2, "expect = ");
+    print_view(col_out->view(),   "result = ");
+  }
+  cudf::test::expect_column_properties_equal(expected_col_out2, col_out->view());
+  cudf::test::expect_columns_equal(expected_col_out2, col_out->view());
+}
```

Comment:
is an empty newline required at end of file?

Reply:
https://unix.stackexchange.com/a/18789/131298

```suggestion

}
```